### DISCUSSION TITLE
I cannot compile application suddenly

### DISCUSSION THREAD

**Discussion by hokonkim** [idaholab/moose#18263] [Q&A General] [answered]
Until yesterday noon, I can recompile my application whenever I want.

But, suddenly with this error message, I cannot recompile the application anymore.

<img width="800" alt="Screen Shot 2021-07-06 at 12 18 18 AM" src="https://user-images.githubusercontent.com/46009457/124541686-eee77300-ddef-11eb-988a-e92661d497f2.png">

I tried to set MOOSE_DIR, first.
(MOOSE_DIR=/anaconda3/envs/moose/ or MOOSE_DIR=/Users/HOKON/projects/moose)

Both of MOOSE_DIR doesn't give me change for fixing the problem, so I unset the MOOSE_DIR currently.

What am I supposed to do fix this problem?

I always appreciate help for MOOSE!

============================================

I searched trobuleshooting in the MOOSE website.

I did "conda update -all" and "make -j4" in my application folder.

<img width="1344" alt="Screen Shot 2021-07-06 at 12 55 35 AM" src="https://user-images.githubusercontent.com/46009457/124544417-142ab000-ddf5-11eb-9fa2-d38c1a3e5435.png">

Now, I got above error.... I am stuck... I think I should wait for a reply from MOOSE staff...

============================================

PS. Because I work on two kinds of operating system (Windows 10 on desktop and MacOS on laptop),

       Usually I edited the source kernel (*.c and *.h) on Windows 10(desktop).

       After that, I copy and paste the edited source kernel to application "src" and "include" folders existed at MacOs(laptop).

       Can this kind of copying and pasting the source codes cause the issue like this?

       I knew the file type? is different between Windows 10(NTFS) and MacOS(..?).

       Even the file format is the same, working different OS for the source code might be an issue for this??

**Comment by GiudGiud** [comment, marked as answer]
Hello

Issue with loading dynamic libraries are commonly due to having an old build lying around. We recommend using `make clobberall` to clean up the directory, then compiing again. 
If that doesn't work, it may be that the cleaning was not enough. In which case, you should commit ALL your chances, then run `git clean -xfd`. 

This type of copy pasting should be fine. You could get issues with line endings if you moved the file, but copy pasting probably works. 
I would recommend using a shared folder for doing this though, or just coding on the mac, for a more efficient workflow. 

Guillaume

**Reply by hokonkim** [reply]
I think now an issue to laptop looks solved.

I tried to recompile the application on HPC.

After "make clobberall" and "make -j4" in my application directory.

1. If I recompile without module (this means, Iuse the local MOOSE in my account),
    it works properly, only for a one time. After the modification of src and include folders, I cannot recompile again.
    What I saw was the same message Iike above.
    "Using HIT from..." and back to command line. It doesn't look like compile.

2. If I recompile the application with the MOOSE module,
   I got error messages like below.

![1](https://user-images.githubusercontent.com/46009457/124640871-03f9eb80-de5c-11eb-9cf7-1faec163eeb6.JPG)

I asked this to HPC staff as well, and wait for his reply. But I don't know why this happens.

For your information, I run a simulation on HPC by using this bash file.

![image](https://user-images.githubusercontent.com/46009457/124641265-808cca00-de5c-11eb-9886-53953a9777f0.png)

Usually, I do not comment out the load module command line.. (2nd one, latest MOOSE)

I knew without loading the MOOSE module, above bash file works.

Let me clear....

"How I fix the compiling problem on HPC in this situation?"

"What is the function of Module on HPC? I thought Module let me use the virtual environment.
 I thought If I load "MOOSE module", then even without local MOOSE file,
 I could make new application (./moose/scirpts/stork.sh app_name), but actually it doesn't work.
 Is there any documents for how module works or what is module on HPC?"

I always appreciate your help!

**Reply by GiudGiud** [reply]
If you have not modified any files after you compiled, it will not recompile. Because everything is already compiled. 
So this may be why you are seeing this "it compiles only once" behavior

Please search the Discussions board for HPC modules related discussions. This has come up recently. 
You cannot switch between MPI distributions between moose and your application, this is likely what is happening here given the error message

**Reply by milljm** [reply]
Also, please refrain from posting pictures of text whenever possible; Search queries cannot find text within a picture (therefor hindering the next user from finding answers if they happen to hit the same error you posted).
